src/instrumenter/instrument.ts

```diff
@@ -490,16 +505,59 @@ export function insertAnnotations(annotations: PropertyMetaData[], ctx: Transpil
 
     const debugInfos = ctx.instrCtx.debugEvents ? getDebugInfo(annotations, ctx) : [];
 
-    const checkStmts: Statement[] = predicates.map(([annotation, predicate], i) => {
-        const event = getAssertionFailedEvent(factory, contract);
+    const checkStmts: Array<[Statement, boolean]> = predicates.map(([annotation, predicate], i) => {
         const dbgInfo = debugInfos[i];
         const emitStmt = dbgInfo !== undefined ? dbgInfo[1] : undefined;
+        const targetIsStmt =
+            annotation.target instanceof Statement ||
+            annotation.target instanceof StatementWithChildren;
 
-        return emitAssert(ctx, predicate, annotation, event, emitStmt);
+        if (annotation.type === AnnotationType.Require) {
+            const reqStmt = factory.makeExpressionStatement(
+                factory.makeFunctionCall(
+                    "<mising>",
+                    FunctionCallKind.FunctionCall,
+                    factory.makeIdentifier("<missing>", "require", -1),
+                    [predicate]
+                )
+            );
+
+            instrCtx.addAnnotationInstrumentation(annotation, reqStmt);
+            instrCtx.addAnnotationCheck(annotation, predicate);
+            return [reqStmt, !targetIsStmt];
+        } else if (annotation.type === AnnotationType.Try) {
+            if (!ctx.hasBinding(ctx.instrCtx.scratchField)) {
+                ctx.addBinding(
+                    ctx.instrCtx.scratchField,
+                    factory.makeElementaryTypeName("<missing>", "uint256")
+                );
+            }
+
+            const lhs = ctx.refBinding(ctx.instrCtx.scratchField);
+            const scratchAssign = factory.makeExpressionStatement(
+                factory.makeAssignment(
+                    "<missing>",
+                    "=",
+                    lhs,
+                    factory.makeLiteral("uint256", LiteralKind.Number, "", "42")
+                )
+            );
+
+            const stmt = factory.makeIfStatement(predicate, scratchAssign);
+            annotation.type === AnnotationType.Try;
+
+            instrCtx.addAnnotationInstrumentation(annotation, stmt);
+            instrCtx.addAnnotationCheck(annotation, predicate);
+
+            return [stmt, !targetIsStmt];
+        } else {
+            const event = getAssertionFailedEvent(factory, contract);
+            return [emitAssert(ctx, predicate, annotation, event, emitStmt), false];
+        }
```

Comment:
Not a blocker, more a reminder for a following cleanups: consider to unnest as each `if`-`else` statement here contains `return` statement. So we may reduce code density in few lines here.

Reply:
Fixed in abe36c2.